File: 14650-걷다보니 신천역 삼 (Small)/main.cpp
#include <iostream>
using namespace std;

int N;
int _count = 0;

void dfs(int start, int digit) {
  if (digit == N) {
    if (start % 3 == 0) _count++;
  } else {
    dfs(start + 0, digit + 1);
    dfs(start + 1, digit + 1);
    dfs(start + 2, digit + 1);
  }
}

int main() {
  cin >> N;
  dfs(1, 1);
  dfs(2, 1);
  cout << _count;
}

File: 14650-걷다보니 신천역 삼 (Small)/main.py
import sys
input = sys.stdin.readline

N = int(input())
count = 0

def dfs(start: int, digit: int):
  if digit == N:
    if start % 3 == 0:
      global count
      count += 1
  else:
    dfs(start+0, digit+1)
    dfs(start+1, digit+1)
    dfs(start+2, digit+1)

dfs(1, 1)
dfs(2, 1)
print(count)